### PR TITLE
add implementation of BiBERT

### PR DESCRIPTION
**Pull request by Macaronlin**
Paddle Implementation of BiBERT (ICLR 2022)

### CODE REVIEW

File: demo/quant/BiBERT/my_task_distill.py [hunk 1]
```diff
@@ -0,0 +1,581 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
```
Comment (qingqing01) [inline demo/quant/BiBERT/my_task_distill.py]: my_task_distill.py   ->  task_distill.py ? remote  prefix "my_"

**Review by qingqing01 — DISMISSED** [review]

File: demo/quant/BiBERT/my_task_distill.py [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+            student_querys = basic._get_attr(student, "query_scores")
+            student_keys = basic._get_attr(student, "key_scores")
+            student_values = basic._get_attr(student, "value_scores")
+            student_reps = basic._get_attr(student, "rep")
+            
+            # print(type(teacher))
+            # print(type(student))
+            # print(len(teacher_querys), len(teacher_keys), len(teacher_values), len(teacher_reps))
+            # print(teacher_querys[0].shape)
+            # print(len(student_querys), len(student_keys), len(student_values), len(student_reps))
```
Comment (qingqing01) [inline demo/quant/BiBERT/my_task_distill.py]: remove debug code

File: demo/quant/BiBERT/my_task_distill.py [hunk 3]
```diff
@@ -0,0 +1,581 @@
...
+                    cls_loss += gt_beta * tmp_loss
+                elif output_mode == "regression":
+                    pass
+                loss = cls_loss
+
+#            if args.intermediate_distill:
+#                loss = cal_intermediate_distill_loss(student, teacher)
+#            else:
+#                loss = ce_loss_fct(logits / args.T,
+#                                   F.softmax(teacher_logits / args.T))
```
Comment (qingqing01) [inline demo/quant/BiBERT/my_task_distill.py]: same as above

**Review by qingqing01 — COMMENTED** [review]
please use pre-commit to format code

File: demo/quant/BiBERT/basic.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+        outs.append(weights)
+    if cache is not None:
+        outs.append(cache)
+    
+    self.query_scores = query_scores
+    self.key_scores = key_scores
+    self.value_scores = value_scores
+    return out if len(outs) == 1 else tuple(outs)
+
+def _BiMultiHeadAttention_forward(self, query, key=None, value=None, attn_mask=None, cache=None):
```
Comment (qingqing01) [inline demo/quant/BiBERT/basic.py]: it seems this code is not  used

File: demo/quant/BiBERT/run_glue.py [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+    else:
+        example = tokenizer(
+            example['sentence1'],
+            text_pair=example['sentence2'],
+            max_seq_len=max_seq_length)
+
+    if not is_test:
+        return example['input_ids'], example['token_type_ids'], label
+    else:
+        return example['input_ids'], example['token_type_ids']
```
Comment (qingqing01) [inline demo/quant/BiBERT/run_glue.py:224]: Please extract the common code between run_glue.py and task_distill.py to a file

File: demo/quant/BiBERT/basic.py (hunk 4 above)
Comment (Macaronlin) [reply demo/quant/BiBERT/basic.py]: fixed

**Comment by htqin** [comment]
Thank you for your review and suggestions! @qingqing01  It seems that the issues mentioned have been fixed, please give a further review :)

**Review by yghstill — APPROVED** [review]

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**